From 362bd90895ff750a5c611286e02198d4ba947edd Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Mon, 7 Apr 2014 20:31:12 -0700
Subject: [PATCH] Ignore non-python files (files not ending in .py).

Also decode the filepath if it is bytes instead of str. This fixes 'Command already registered' errors when reloading plugins.

diff --git a/core/loader.py b/core/loader.py
index 41f7a67..579b28d 100644
--- a/core/loader.py
+++ b/core/loader.py
@@ -59,11 +59,19 @@ class PluginLoader(object):
         :type rebuild: bool
         """
         filename = os.path.basename(path)
-        title = os.path.splitext(filename)[0]
+        if isinstance(filename, bytes):
+            # makes sure that the filename is a 'str' object, not a 'bytes' object
+            filename = filename.decode()
+        file_split = os.path.splitext(filename)
+        title = file_split[0]
+        extension = file_split[1]
+        if extension != ".py":
+            # ignore non-python plugin files
+            return
 
         disabled = self.bot.config.get('disabled_plugins', [])
         if title in disabled:
-            self.bot.logger.info("Did not load plugins from: {} (plugin disabled)".format(filename))
+            self.bot.logger.info("Not loading plugin {}: plugin disabled".format(filename))
             return
 
         # compile the file and eval it in a namespace
@@ -106,6 +114,21 @@ class PluginLoader(object):
         :type path: str
         """
         filename = os.path.basename(path)
+        if isinstance(filename, bytes):
+            # makes sure that the filename is a 'str' object, not a 'bytes' object
+            filename = filename.decode()
+        file_split = os.path.splitext(filename)
+        title = file_split[0]
+        extension = file_split[1]
+        if extension != ".py":
+            # ignore non-python plugin files
+            return
+
+        disabled = self.bot.config.get('disabled_plugins', [])
+        if title in disabled:
+            # this plugin hasn't been loaded, so no need to unload it
+            return
+
         self.bot.logger.info("Unloading plugins from: {}".format(filename))
 
         # remove plugins loaded from this file
-- 
1.9.1

